docs: align self-managed infrastructure docs with e2e worfklow (#2525)
edgelesssys/constellation: docs/docs/workflows/create.md

The Constellation [apply step](#the-apply-step) requires the already created `constellation-config.yaml` and the `constellation-state.yaml`.
Create the `constellation-state.yaml` using the output from the Terraform state and the `constellation-conf.yaml`:

> <tabs groupId="provider-output">
> <tabItem value="aws-output" label="AWS">

Reviewer: ```suggestion
<tabs groupId="csp">
<tabItem value="aws" label="AWS">
```
groupId and value names must always be the same. Then, if you switch a tab in on place, all other places are switched, too.

> <tabItem value="aws-output" label="AWS">

Reviewer: there's another groupId above that you should rename from `provider` to `csp`

```bash
yq eval '.version ="v1"' --inplace constellation-state.yaml
yq eval ".infrastructure.initSecret =\"$(terraform output initSecret | jq -r | tr -d '\n' | hexdump -ve '/1 "%02x"' && echo '')\"" --inplace constellation-state.yaml
yq eval ".infrastructure.clusterEndpoint =\"$(terraform output out_of_cluster_endpoint | jq -r)\"" --inplace constellation-state.yaml
yq eval ".infrastructure.inClusterEndpoint =\"$(terraform output in_cluster_endpoint | jq -r)\"" --inplace constellation-state.yaml
yq eval ".infrastructure.ipCidrNode =\"$(terraform output ip_cidr_nodes | jq -r)\"" --inplace constellation-state.yaml
yq eval ".infrastructure.uid =\"$(terraform output uid | jq -r)\"" --inplace constellation-state.yaml
yq eval ".infrastructure.name =\"$(terraform output name | jq -r)\"" --inplace constellation-state.yaml
yq eval ".infrastructure.apiServerCertSANs =$(terraform output -json api_server_cert_sans)" --inplace constellation-state.yaml
```

</tabItem>

> <tabItem value="azure-output" label="Azure">

Reviewer: ```suggestion
<tabItem value="azure" label="Azure">
```

:::info

  If the enforcement policy is set to `MAAFallback` in `constellation-config.yaml`, a manual update to the MAA provider's policy is necessary.

>   The update can be applied with the following commands, where `<VERSION>` is the version of Constellation that should be set up. (e.g. `v2.12.0`)

Reviewer: ```suggestion
  You can apply the update with the following commands, where `<VERSION>` is the version of Constellation that should be set up. (e.g. `v2.12.0`)
```

  ```bash

>   https://github.com/edgelesssys/constellation/tree/<VERSION>

Reviewer 2: ```suggestion
  git clone --branch <VERSION> https://github.com/edgelesssys/constellation
```

  cd constellation/hack/maa-patch

>   go run . $(terraform output attestationURL | jq -r)

Reviewer: Azure CLI doesn't support this operation, does it?

Author: Theoretically, it does, but it would require a fairly long script that we would have to potentially maintain along the code in our CLI, why I opted to share the code and create a tool instead.

yq eval ".infrastructure.uid =\"$(terraform output uid | jq -r)\"" --inplace constellation-state.yaml
yq eval ".infrastructure.name =\"$(terraform output name | jq -r)\"" --inplace constellation-state.yaml
yq eval ".infrastructure.apiServerCertSANs =$(terraform output -json api_server_cert_sans)" --inplace constellation-state.yaml
yq eval ".infrastructure.azure.resourceGroup =\"$(terraform output resource_group | jq -r)\"" --inplace constellation-state.yaml
yq eval ".infrastructure.azure.subscriptionID =\"$(terraform output subscription_id | jq -r)\"" --inplace constellation-state.yaml
yq eval ".infrastructure.azure.networkSecurityGroupName =\"$(terraform output network_security_group_name | jq -r)\"" --inplace constellation-state.yaml
yq eval ".infrastructure.azure.loadBalancerName =\"$(terraform output loadbalancer_name | jq -r)\"" --inplace constellation-state.yaml
yq eval ".infrastructure.azure.userAssignedIdentity =\"$(terraform output user_assigned_identity_client_id | jq -r)\"" --inplace constellation-state.yaml
yq eval ".infrastructure.azure.attestationURL =\"$(terraform output attestationURL | jq -r)\"" --inplace constellation-state.yaml
```

</tabItem>

> <tabItem value="gcp-output" label="GCP">

Reviewer: ```suggestion
<tabItem value="gcp" label="GCP">
```

They contain a minimum configuration for the resources necessary to run a Constellation cluster on the corresponding CSP. From this base, you can now add, edit, or substitute resources per your own requirements with the infrastructure
management tooling of your choice. You need to keep the essential functionality of the base configuration in order for your cluster to function correctly.

:::info

  On Azure, if the enforcement policy is set to `MAAFallback` in `constellation-config.yaml`, a manual update to the MAA provider's policy is necessary.

>   The update can be applied with the following commands, where `<VERSION>` is the version of Constellation that should be set up. (e.g. `v2.12.0`)

Reviewer: ```suggestion
  You can apply the update with the following commands, where `<VERSION>` is the version of Constellation that should be set up. (e.g. `v2.12.0`)
```

  ```bash

>   https://github.com/edgelesssys/constellation/tree/<VERSION>

Reviewer 2: ```suggestion
  git clone --branch <VERSION> https://github.com/edgelesssys/constellation
```
